src/Rib/Shake.hs

```diff
@@ -37,22 +36,37 @@ import Data.Typeable
 import Development.Shake
 import Development.Shake.FilePath
 import Development.Shake.Forward (cacheAction)
+import System.Directory (createDirectoryIfMissing)
 import Lucid (Html)
 import qualified Lucid
 import Text.Pandoc (Pandoc (Pandoc), PandocIO, ReaderOptions)
 
-import Rib.App (ribInputDir, ribOutputDir)
 import qualified Rib.Pandoc
 
--- FIXME: Auto create ./b directory
+newtype Dirs = Dirs (FilePath, FilePath)
+
+getDirs :: Action (FilePath, FilePath)
+getDirs = getShakeExtra >>= \case
+  Just (Dirs d) -> return d
+  Nothing       -> fail "Input output directories are not initialized"
```

Comment:
The 2d indentation here should be removed, to maintain a consistent coding style throughout the project.

(Normally I just run `stylish-haskell` using [this config](https://github.com/srid/Taut/blob/master/.stylish-haskell.yaml)).

Reply:
Fixed.